src/components/calculator/Calculate.js

```diff
@@ -0,0 +1,74 @@
+/* eslint-disable */
+import { useState } from 'react';
+import classes from './Calculate.module.css';
+import calculate from '../../logic/calculate.js';
+import Button from '../UI/Button.js';
+
+const Calculate = (props) => {
+  const [next, setNext] = useState(0);
+
+  const clickHandler = (e) => {
+    setNext((state) => {
+      return calculate(state, e.target.textContent);
+    });
+  };
+
+  return (
+    <div className={props.className}>
+      <div className={classes.calculator}>
+        <div className={classes.row1}>
+          {next.next ?? next.total ?? next.operation ?? 0}
+        </div>
+        <div className={classes.row}>
+          <Button onClick={clickHandler}>AC</Button>
+          <Button onClick={clickHandler}>+/-</Button>
+          <Button onClick={clickHandler}>%</Button>
+          <Button className={` ${classes.btn_orange}`} onClick={clickHandler}>
+            +
+          </Button>
+        </div>
+        <div className={classes.row}>
+          <Button onClick={clickHandler}>7</Button>
+          <Button onClick={clickHandler}>8</Button>
+          <Button onClick={props.clickHandler}>9</Button>
```

Comment:
- [ ] Button 9 does not work; when I click it nothing is displayed. Kindly fix that. You should remove the `props` here.

  ```
  <Button onClick={clickHandler}>9</Button>
  ```

Reply:
Oh great! Thank you very much.